airflow/contrib/operators/gcp_transfer_operator.py

```diff
@@ -0,0 +1,753 @@
+
+    :param operation_name: (Required) Name of the transfer operation.
+    :type operation_name: str
+    :param gcp_conn_id: The connection ID used to connect to Google Cloud Platform.
+    :param api_version: API version used (e.g. v1).
+    :type api_version: str
+    :type gcp_conn_id: str
+    """
+
+    # [START gcp_transfer_operation_resume_template_fields]
+    template_fields = ('operation_name', 'gcp_conn_id', 'api_version')
+    # [END gcp_transfer_operation_resume_template_fields]
+
+    @apply_defaults
+    def __init__(self, operation_name, gcp_conn_id='google_cloud_default', api_version='v1', *args, **kwargs):
+
+        self.operation_name = operation_name
+        self.gcp_conn_id = gcp_conn_id
+        self.api_version = api_version
+        self._validate_inputs()
+        super(GcpTransferServiceOperationResumeOperator, self).__init__(*args, **kwargs)
+
+    def _validate_inputs(self):
+        if not self.operation_name:
+            raise AirflowException("The required parameter 'operation_name' is empty or None")
+
+    def execute(self, context):
+        hook = GCPTransferServiceHook(api_version=self.api_version, gcp_conn_id=self.gcp_conn_id)
+
+        hook.resume_transfer_operation(operation_name=self.operation_name)
+
+
+class GcpTransferServiceOperationCancelOperator(BaseOperator):
+    """
+    Cancels a transfer operation in Google Storage Transfer Service.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:GcpTransferServiceOperationCancelOperator`
+
+    :param operation_name: (Required) Name of the transfer operation.
+    :type operation_name: str
+    :param api_version: API version used (e.g. v1).
+    :type api_version: str
+    :param gcp_conn_id: The connection ID used to connect to Google
+        Cloud Platform.
+    :type gcp_conn_id: str
+    """
+
+    # [START gcp_transfer_operation_cancel_template_fields]
+    template_fields = ('operation_name', 'gcp_conn_id', 'api_version')
+    # [END gcp_transfer_operation_cancel_template_fields]
+
+    @apply_defaults
+    def __init__(self, operation_name, api_version='v1', gcp_conn_id='google_cloud_default', *args, **kwargs):
+        super(GcpTransferServiceOperationCancelOperator, self).__init__(*args, **kwargs)
+        self.operation_name = operation_name
+        self.api_version = api_version
+        self.gcp_conn_id = gcp_conn_id
+        self._validate_inputs()
+
+    def _validate_inputs(self):
+        if not self.operation_name:
+            raise AirflowException("The required parameter 'operation_name' is empty or None")
+
+    def execute(self, context):
+        hook = GCPTransferServiceHook(api_version=self.api_version, gcp_conn_id=self.gcp_conn_id)
+        hook.cancel_transfer_operation(operation_name=self.operation_name)
+
+
+class S3ToGoogleCloudStorageTransferOperator(BaseOperator):
```

Comment:
I did not write this operator. I just copied it here and made changes necessary for proper functioning.We tried to limit the changes to only those necessary. Otherwise. This operator has no common logic with this operator. Operators are built by compositions, not inheritance.  Common logic is in a different class - hook.  Inheritance would not bring any benefits in this case. Only it would raise the complexity of the code.